### ISSUE TITLE
redundant_string_enum_value should only generate a warning if all cases in the enum are redundant

### ISSUE THREAD

**Issue by julsh** [realm/SwiftLint#1820]
Version 0.22.0

consider the following enum:
```
enum Cats: String {
    case tiger = "tiger"
    case leopard = "leopard"
    case snowLeopard = "snow_leopard"
    case lion = "lion"
    case mountainLion = "mountain_lion"
}
```

The `redundant_string_enum_value` rule would throw a warning on the cases tiger, leopard and lion. To fix this, one would have to adjust the enum to this:
```
enum Cats: String {
    case tiger
    case leopard
    case snowLeopard = "snow_leopard"
    case lion
    case mountainLion = "mountain_lion"
}
```

I'd argue that this mixed-style enum where some cases have an explicit String value assigned and some don't is actually less clean and would prefer the first version with all cases having an expclicit value.

My suggestion would be to make the `redundant_string_enum_value` take into account the entire enum and only throw a warning if the string value for every single case in the enum is redundant.

**Comment by marcelofabri** [comment]
The rule shouldn't trigger in that case, and it doesn't for me:

```
$ cat file.swift 
enum Cats: String {
    case tiger = "tiger"
    case leopard = "leopard"
    case snowLeopard = "snow_leopard"
    case lion = "lion"
    case mountainLion = "mountain_lion"
}
$ swiftlint lint --path file.swift --no-cache --enable-all-rules
Loading configuration from '.swiftlint.yml'
Linting Swift files at path file.swift
Linting 'file.swift' (1/1)
/Users/marcelofabri/projetos/SwiftLint/file.swift:1:1: warning: Explicit Top Level ACL Violation: Top-level declarations should specify Access Control Level keywords explicitly. (explicit_top_level_acl)
/Users/marcelofabri/projetos/SwiftLint/file.swift:1: warning: File Header Violation: Header comments should be consistent with project patterns. (file_header)
Done linting! Found 2 violations, 0 serious in 1 file.
```

**Comment by julsh** [comment]
ah interesting. i just came up with this example on the fly and didn't actually evaluate it. it triggered in a switch case similar to this one, but maybe there's something else going on. i'll double-check and get back to you.

**Comment by julsh** [comment]
My apologies, I checked again and it was actually triggering on a different enum in the file that I was having trouble with - and this enum in fact had all redundant cases. Guess I should have been a bit more thorough before opening an issue. Whoops :)

**Comment by marcelofabri** [comment]
No worries, thanks for the update!
